daemon/api.go

```diff
@@ -2801,6 +2811,53 @@ var (
 	statePendingWarnings = (*state.State).PendingWarnings
 )
 
+type postModelData struct {
+	NewModel string `json:"new-model"`
+}
+
+func postModel(c *Command, r *http.Request, _ *auth.UserState) Response {
+	defer r.Body.Close()
+	var data postModelData
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&data); err != nil {
+		return BadRequest("cannot decode request body into remodel operation: %v", err)
+	}
+	rawNewModel, err := asserts.Decode([]byte(data.NewModel))
+	if err != nil {
+		return BadRequest("cannot decode new model assertion: %v", err)
+	}
+	newModel, ok := rawNewModel.(*asserts.Model)
+	if !ok {
+		return BadRequest("new model is not a model assertion: %v", newModel.Type())
+	}
+
+	st := c.d.overlord.State()
+	st.Lock()
+	defer st.Unlock()
+
+	tss, err := devicestateRemodel(st, newModel)
```

Comment:
do you want to add checking for seeded there in this PR? a separate PR? or the current follow up one?

Reply:
Its really small so I added it to this PR.